packages/story-editor/src/components/videoTrim/provider.js

```diff
@@ -40,6 +48,40 @@ function VideoTrimProvider({ children }) {
     resetOffsets,
   } = useVideoNode();
 
+  /**
+   * Converts milliseconds to HH:MM:SS.SSS format.
+   *
+   * @type {function(*): string}
+   */
+  const msToHMS = useCallback((ms) => {
+    let seconds = ms / 1000;
+    const hours = parseInt(seconds / 3600);
+    seconds = seconds % 3600;
+    const minutes = parseInt(seconds / 60);
+    seconds = seconds % 60;
+    return `${hours}:${minutes}:${seconds}`;
+  }, []);
+
+  const performTrim = useCallback(() => {
+    const { resource } = selectedElements[0];
+    if (!resource) {
+      return;
+    }
+    trimExistingVideo({
+      resource,
+      start: msToHMS(startOffset),
+      end: msToHMS(endOffset),
+    });
+    toggleTrimMode();
```

Comment:
It seems to me that exiting the trim mode when starting trimming makes the most sense. Otherwise, the trim bar should be disabled anyway in trim mode and then once finished updated to the new values of the trimmed video. Then the user can continue doing other stuff and the Trim button is disabled with the Loading circle to indicate the processing happening.